cluster-autoscaler/simulator/basic_cluster_snapshot.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	schedulerlisters "k8s.io/kubernetes/pkg/scheduler/listers"
+	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
+)
+
+// BasicClusterSnapshot is simple, reference implementation of ClusterSnapshot.
+// It is inefficient. But hopefully bug-free and good for initial testing.
```

Comment:
> bug-free

That is one a brave statement

Reply:
Yeah - would be nice to have some tests. Kinda hoped that @aleksandra-malinowska will add those for more efficient implementations. And then I could also could test basic one with those. Not sure if that happened :)

Reply:
Added just one correctness test for now, but planning more. BTW the other PR is at https://github.com/losipiuk/autoscaler/pull/1